### PR TITLE
AA/MIN-455 mnt pallet renaming

### PR DESCRIPTION
**Pull request by ales-by**
**Description:**

 mnt pallet renaming

**Changes in storage or API:**
*RPC changed:*
get_mnt_borrow_and_supply_rates -> get_pool_mnt_borrow_and_supply_rates
get_unclaimed_mnt_balance  ->  get_user_total_unclaimed_mnt_balance

*Storage*
MntClaimThreshold -> MntClaimThresholdStorage
MntSupplierIndex -> MntSupplierIndexStorage
MntBorrowerIndex -> MntBorrowerIndexStorage
MntBorrowerIndex -> MntBorrowerIndexStorage
MntAccrued -> MntAccruedStorage

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

### CODE REVIEW

**Review by Jekshmek — APPROVED** [review]

**Review by Jekshmek — APPROVED** [review]

**Comment by lut1k** [comment]
Don't forget to describe the changed storage in the API documentation. The frontend will need to fix the queries.

File: pallets/mnt-token/src/lib.rs [hunk 1]
```diff
@@ -294,37 +295,43 @@ impl<T: Config> Pallet<T> {
...
 	}
 }
 
 impl<T: Config> MntManager<T::AccountId> for Pallet<T> {
 	/// Update mnt supply index for pool
 	/// Do nothing in case if distribution is off or index have already been updated in this block.
-	fn update_mnt_supply_index(underlying_id: CurrencyId) -> DispatchResult {
+	///
+	/// - `pool_id`: id of the pool to update index
+	fn pool_update_mnt_supply_index(pool_id: CurrencyId) -> DispatchResult {
```
Comment (lut1k) [inline pallets/mnt-token/src/lib.rs:320]:
```suggestion
	fn update_pool_mnt_supply_index(pool_id: CurrencyId) -> DispatchResult {
```
Maybe this is better?

File: pallets/mnt-token/src/lib.rs [hunk 2]
```diff
@@ -359,28 +364,30 @@ impl<T: Config> MntManager<T::AccountId> for Pallet<T> {
...
+		MntPoolStateStorage::<T>::insert(pool_id, pool_state);
 		Ok(())
 	}
 
 	/// Update mnt borrow index for pool
 	/// Do nothing in case if distribution is off or index have already been updated in this block.
-	fn update_mnt_borrow_index(underlying_id: CurrencyId) -> DispatchResult {
+	///
+	/// - `pool_id`: id of the pool to update index
+	fn pool_update_mnt_borrow_index(pool_id: CurrencyId) -> DispatchResult {
```
Comment (lut1k) [inline pallets/mnt-token/src/lib.rs:375]: same

**Review by lut1k — DISMISSED** [review]

File: pallets/mnt-token/src/lib.rs (hunk 2 above)
Comment (ales-by) [reply pallets/mnt-token/src/lib.rs]: done

**Review by lut1k — APPROVED** [review]

File: pallets/mnt-token/src/lib.rs [hunk 3]
```diff
@@ -334,21 +341,19 @@ impl<T: Config> MntManager<T::AccountId> for Pallet<T> {
...
+		let wrapped_asset_id = pool_id.wrapped_asset().ok_or(Error::<T>::NotValidUnderlyingAssetId)?;
 
 		let block_delta_as_u128 = TryInto::<u128>::try_into(block_delta).or(Err(Error::<T>::InternalError))?;
 
 		let mnt_accrued = mnt_speed
 			.checked_mul(block_delta_as_u128)
 			.ok_or(Error::<T>::NumOverflow)?;
 
-		let total_tokens_supply = T::MultiCurrency::total_issuance(wrapped_asset_id);
+		let pool_token_supply_wrapped = T::MultiCurrency::total_issuance(wrapped_asset_id);
```
Comment (Apollo50) [inline pallets/mnt-token/src/lib.rs:352]:
```suggestion
		let pool_supply_wrapped = T::MultiCurrency::total_issuance(wrapped_asset_id);
```

File: pallets/mnt-token/src/lib.rs [hunk 4]
```diff
@@ -473,30 +478,27 @@ impl<T: Config> MntManager<T::AccountId> for Pallet<T> {
...
-		MntBorrowerIndex::<T>::insert(underlying_id, borrower, pool_borrow_state.mnt_distribution_index);
+		MntBorrowerIndexStorage::<T>::insert(pool_id, borrower, pool_borrow_state.mnt_distribution_index);
 		if borrower_index.is_zero() {
 			// This is first interaction with protocol
 			return Ok(Balance::zero());
 		}
 
-		let borrow_balance = T::ControllerManager::borrow_balance_stored(&borrower, underlying_id)?;
-		let pool_borrow_index = T::LiquidityPoolsManager::get_pool_borrow_index(underlying_id);
+		let borrow_balance = T::ControllerManager::borrow_balance_stored(&borrower, pool_id)?;
```
Comment (Apollo50) [inline pallets/mnt-token/src/lib.rs:500]:
```suggestion
		let user_borrow_underlying = T::ControllerManager::borrow_balance_stored(&borrower, pool_id)?;
```

**Review by Apollo50 — APPROVED** [review]

**Review by aparkalov — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Pull request merged**